src/entities/createPool/types.ts

```diff
@@ -0,0 +1,33 @@
+import { Address, Hex } from 'viem';
+
+export interface CreatePoolBase {
+    buildCall(input: CreatePoolInput): CreatePoolBuildCallOutput;
+}
+
+export type CreatePoolInput = CreateWeightedPoolInput;
+
+export type CreateWeightedPoolInput = {
+    name?: string;
+    symbol: string;
+    tokens: Address[];
+    weights: { tokenAddress: Address; weight: string }[];
+    rateProviders: { tokenAddress: Address; rateProviderAddress: Address }[];
+    swapFee: string;
+    poolOwnerAddress: Address;
+    salt?: Hex;
+};
+
+export type CreatePoolBuildCallOutput = {
+    call: Hex;
+};
+
+export type CreateWeightedPoolArgs = [
```

Comment:
This is still not following the pattern - it should be `CreatePoolWeigthedInput` and `CreatePoolWeightedArgs`.
This will make more sense when you start implementing Composable Stable pool create and have `CreatePoolComposableStableInput` and `CreatePoolComposableStableArgs`.
It's easier to follow when the prefixes are the same.